extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/Util.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.binding.onewire.internal;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.measure.quantity.Dimensionless;
+import javax.measure.quantity.Temperature;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.library.dimension.Density;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.types.State;
+
+/**
+ * The {@link Util} is a set of helper functions
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@NonNullByDefault
+public class Util {
+
+    /**
+     * convert a number to a list of booleans
+     *
+     * @param decimal
+     * @return
+     */
+    public static List<Boolean> decimalTypeToBooleanList(DecimalType decimal) {
+        List<Boolean> returnValues = new ArrayList<Boolean>();
+        int value = decimal.intValue();
+        for (int i = 0; i < 8; i++) {
+            returnValues.add((value & (1 << i)) > 0);
+        }
+        return returnValues;
+    }
+
+    /**
+     * convert measured value to temperature with offset
+     *
+     * @param temperature measured value
+     * @param offset offset to apply
+     * @return originalState if not DecimalType, temperature otherwise
+     * @throws OwException in case of conversion error
+     */
+    public static QuantityType<Temperature> convertTemperature(QuantityType<Temperature> temperature,
+            QuantityType<Temperature> offset) throws OwException {
+        QuantityType<Temperature> returnTemperature = temperature.toUnit(offset.getUnit());
+        if (returnTemperature == null) {
+            throw new OwException("error while converting measured value");
+        }
+        returnTemperature = returnTemperature.add(offset).toUnit(SIUnits.CELSIUS);
+        if (returnTemperature == null) {
+            throw new OwException("error while adding offset");
+        }
+        return returnTemperature;
+    }
+
+    /**
+     * calculate absolute humidity in g/m³ from measured values
+     *
+     * @param temperature the measured temperature
+     * @param relativeHumidity the measured relative humidity
+     * @return the corresponding absolute humidity
+     */
+    public static State calculateAbsoluteHumidity(QuantityType<Temperature> temperature,
```

Comment:
nice. In the light of #4748  maybe we can extract this when a solution is found.